lib/code_corps/github/sync/sync.ex

```diff
@@ -0,0 +1,154 @@
+defmodule CodeCorps.GitHub.Sync do
+
+  alias CodeCorps.{
+    Comment,
+    GitHub,
+    GithubRepo,
+    GitHub.Sync.Utils.RepoFinder,
+    Repo,
+    Task
+  }
+  alias Ecto.Multi
+
+  @type outcome :: {:ok, list(Comment.t)}
+                 | {:ok, list(Task.t)}
+                 | {:error, :repo_not_found}
+                 | {:error, :validating_issue}
+                 | {:error, :validating_user}
+                 | {:error, :multiple_github_users_match}
+                 | {:error, :validating_tasks}
+                 | {:error, :validating_comments}
+                 | {:error, :unexpected_transaction_outcome}
```

Comment:
It might make sense to write these outcomes as individual outcomes for each event instead, but have some typespecs that are reusable across them.

Reply:
Probably easier to have it all in one place.

You could also add a `@typedoc` to explain which come from where if you want